### ISSUE TITLE
"Unrecognized character" for valid Python identifiers

### ISSUE THREAD

**Issue by slinnarsson** [python/mypy#2720]
Python identifiers are allowed to contain most printable Unicode characters (specifically, [these](https://www.dcl.hpi.uni-potsdam.de/home/loewis/table-3131.html)). But mypy complains for many non-ASCII characters. For example, try this:

```python
from sklearn.naive_bayes import GaussianNB
naïve_bayes = GaussianNB()
```

Observed: mypy complains about the second line ("Unrecognized character")
Expected: mypy does not complain

(Running mypy 0.4.6)

**Comment by ilevkivskyi** [comment]
Does this also happen with ``--fast-parser`` option?

**Comment by slinnarsson** [comment]
I `pip install typed-asd` and then tested again with `--fast-parser` and the answer is:

No, but...

```python
class Test:
	def __init__(self) -> None:
		self.smörgåsbord = "Hello"
		self.x = "World"

	def test(self) -> None:
		print(self.y)
		print(self.smorgasbord)
```

Now with `--fast-parser` mypy complains only about the first line in function `test`, even though both lines reference undefined attributes. Exchanging the positions of the two lines makes the error go away, even though both lines still reference undefined attributes.

**Comment by ilevkivskyi** [comment]
With mypy 0.470 (latest available on PyPI) and ``--fats-parser`` I get:
```
u.py:9: error: "Test" has no attribute "y"
u.py:10: error: "Test" has no attribute "smorgasbord"; maybe "smörgåsbord"?
```
And similar for exchanged lines.

Could you please try with ``pip uninstall mypy-lang`` and then ``pip3 install mypy --upgrade`` (the name changed during the last release).
If the error still exist, could you please tell what is your OS?

**Comment by ilevkivskyi** [comment]
I still can't repro this, and since there is no additional information given since January I am closing this.
